backend/connector_v2/connector_instance_helper.py

```diff
@@ -0,0 +1,331 @@
+            connector_name=connector_name,
+            workflow=workflow,
+            created_by=user,
+            connector_id=gcs_id,
+            connector_metadata=connector_metadata__output,
+            connector_type=ConnectorInstance.ConnectorType.OUTPUT,
+            connector_mode=ConnectorInstance.ConnectorMode.FILE_SYSTEM,
+        )
+        logger.info("Connectors added successfully.")
+
+    @staticmethod
+    def get_connector_instances_by_workflow(
+        workflow_id: str,
+        connector_type: tuple[str, str],
+        connector_mode: Optional[tuple[int, str]] = None,
+        values: Optional[list[str]] = None,
+        connector_name: Optional[str] = None,
+    ) -> list[ConnectorInstance]:
+        """Method to get connector instances by workflow.
+
+        Args:
+            workflow_id (str)
+            connector_type (tuple[str, str]): Specifies input/output
+            connector_mode (Optional[tuple[int, str]], optional):
+                Specifies database/file
+            values (Optional[list[str]], optional):  Defaults to None.
+            connector_name (Optional[str], optional):  Defaults to None.
+
+        Returns:
+            list[ConnectorInstance]
+        """
+        logger.info(f"Setting connector mode to {connector_mode}")
+        filter_params: dict[str, Any] = {
+            "workflow": workflow_id,
+            "connector_type": connector_type,
+        }
+        if connector_mode is not None:
+            filter_params["connector_mode"] = connector_mode
+        if connector_name is not None:
+            filter_params["connector_name"] = connector_name
+
+        connector_instances = ConnectorInstance.objects.filter(**filter_params).all()
+        logger.debug(f"Retrieved connector instance values {connector_instances}")
+        if values is not None:
+            filtered_connector_instances = connector_instances.values(*values)
+            logger.info(
+                f"Returning filtered \
+                    connector instance value {filtered_connector_instances}"
+            )
+            return list(filtered_connector_instances)
+        logger.info(f"Returning connector instances {connector_instances}")
+        return list(connector_instances)
+
+    @staticmethod
+    def get_connector_instance_by_workflow(
+        workflow_id: str,
+        connector_type: tuple[str, str],
+        connector_mode: Optional[tuple[int, str]] = None,
+        connector_name: Optional[str] = None,
+    ) -> Optional[ConnectorInstance]:
+        """Get one connector instance.
+
+            Use this method if the connector instance is unique for \
+                filter_params
+        Args:
+            workflow_id (str): _description_
+            connector_type (tuple[str, str]):  Specifies input/output
+            connector_mode (Optional[tuple[int, str]], optional).
+                Specifies database/filesystem
+            values (Optional[list[str]], optional).
```

Comment:
@muhammad-ali-e need to remove this? This is not available in args.

Reply:
Updated